db/types/mathesar_char.py

```diff
@@ -0,0 +1,22 @@
+from sqlalchemy.dialects.postgresql import CHAR as SA_CHAR
+from sqlalchemy.types import TypeDecorator
+from sqlalchemy.ext.compiler import compiles
+
+from db.types.base import PostgresType
+
+
+class MATHESAR_CHAR(TypeDecorator):
+    impl = SA_CHAR
+
+    @classmethod
+    def __str__(cls):
+        return cls.__name__
+
+
+@compiles(MATHESAR_CHAR, 'postgresql')
+def _compile_mathesar_char(element, compiler, **kw):
+    unchanged_compiled_string = compiler.visit_VARCHAR(element, **kw)
```

Comment:
@dmos62 @mathemancer Here I want to use `String` class of Python so can you please guide me how to compile this thing as @dmos62 demonstrated in the ticket for varchar.

Reply:
The `unchanged_id` should still be `VARCHAR`. The changed ID should be `"char"`. Note the casing and quotes. Those are required.

Reply:
@mathemancer I am planning to get the value of changes_id via POSTGRESQL.MATHSAR_CHAR so I have made those changes in the required files. Can you please share some code snippets how to write such tests as I went through the tests of API layers but unable to make any progress?

Reply:
See the tests in `db/tests/types/test_money.py` for examples of making sure creation and reflection are working correctly.